skstore/src/Skdb.sk

```diff
@@ -0,0 +1,182 @@
+module alias P = SQLParser;
+
+module SKDBStore;
+
+class TypeError(msg: String) extends Exception {
+  fun getMessage(): String {
+    this.msg
+  }
+}
+
+class Filter(query: String, params: ?SKJSON.CJObject) uses Orderable {
+  static fun fromJSON(json: SKJSON.CJSON): ?Filter {
+    json match {
+    | obj @ SKJSON.CJObject _ ->
+      SKJSON.getString(obj, "query").map(query ->
+        Filter(query, SKJSON.getObject(obj, "params"))
+      )
+    | _ -> None()
+    }
+  }
+}
+
+class FilterKey(value: Filter) extends SKStore.Key
+
+class FilterFile(value: Filter) extends SKStore.File
+
+class JSONHandle(
+  value: SKStore.EHandle<SKStore.JSONID, SKStore.JSONFile>,
+) extends SKStore.File
+
+fun mapTable(
+  table: String,
+  name: String,
+  context: mutable SKStore.Context,
+  compute: (
+    mutable SKStore.Context,
+    mutable SKStore.TWriter<SKStore.JSONID, SKStore.JSONFile>,
+    SKDB.RowValues,
+  ) ~> void,
+): String {
+  dirDescr = SKDB.getTable(context, 0, P.Name::create(table));
+  handle = SKStore.EHandle(
+    SKDB.RowKey::keyType,
+    SKDB.RowValues::type,
+    dirDescr.dirName,
+  );
+  dirName = SKStore.subDirName(context, name);
+  _ = handle.map(
+    SKStore.JSONID::keyType,
+    SKStore.JSONFile::type,
+    context,
+    dirName,
+    (context, writer, _, rit) ~> {
+      compute(context, writer, rit.first)
+    },
+  );
+  dirName.toString();
+}
+
+fun toTable(
+  handle: SKStore.EHandle<SKStore.JSONID, SKStore.JSONFile>,
+  table: String,
+  context: mutable SKStore.Context,
+  conv: (
+    SKStore.JSONID,
+    mutable SKStore.NonEmptyIterator<SKStore.JSONFile>,
+  ) ~> SKJSON.CJSON,
+): void {
+  dirDescr = SKDB.getTable(context, 0, P.Name::create(table));
+  dirName = SKStore.subDirName(context, "outputs/" + table);
+  converted = handle.map(
+    SKDB.RowKey::keyType,
+    SKDB.RowValues::type,
+    context,
+    dirName,
+    (context, writer, key, it) ~> {
+      computed = conv(key, it);
+      rowValues = toRowValues(context, dirDescr, computed);
+      rowKey = SKDB.RowKey(rowValues, dirDescr.kinds);
+      writer.set(rowKey, rowValues)
+    },
+  );
+  context.maybeGetEagerDir(converted.getDirName()).each(dir ->
+    dir.copyTo(context, dirDescr.dirName)
+  )
+}
+
+fun decodeParams(input: SKJSON.CJObject): Map<String, P.Value> {
```

Comment:
Indeed, it's the version that uses SKJSON, and the code that call it no more in that version. We probably need to use only SKJSON in SKDB to prevent this kind of thing.